Sources/macro.js

```diff
@@ -1200,6 +1200,37 @@ export function proxyPropertyState(
   }
 }
 
+// ----------------------------------------------------------------------------
+// keystore(publicAPI, model, initialKeystore)
+//
+//    - initialKeystore: Initial keystore. This can be either a Map or an
+//      object.
+//
+// Generated API
+//  setKey(key, value) : undefined
+//  getKey(key) : mixed
+//  getAllKeys() : [mixed]
+//  deleteKey(key) : Boolean
+// ----------------------------------------------------------------------------
+
+export function keystore(publicAPI, model, initialKeystore = {}) {
+  const store = new Map();
```

Comment:
Do we really need a Map? Would it be good enough to use `{}` instead?

Do we want it stored on the `model`?

Reply:
I tend to use Map for key/value stores. Plus, if someone really wants to use an object as a key, they can with Map. :P Is there a reason to use `{}` instead?

Yes, storing on the model would be nice. I'll do that.

Reply:
I understand, but then you break the current pattern that enable serialization on any vtkObject:

```
const str = obj.getState();
const newObj = vtk(str);
```

Also you would have to do something like

```
if (!model.keystore) {
   model.keystore = {};
}
```

Reply:
Oh, didn't know about object serialization. In that case yes, I'll switch to using an object.